qiskit/transpiler/preset_passmanagers/default.py

```diff
@@ -81,6 +82,9 @@ def default_pass_manager(basis_gates, coupling_map, initial_layout,
     else:
         simplification_passes = [CXCancellation()]
 
+    if 'reset' in basis_gates:
```

Comment:
can you remove this? this optimization is not related to having reset in the basis.

Reply:
Otherwise the pass does nothing, because it searches for the `Reset` gate after unrolling.

Reply:
I think there's a confusion here between `basis_gates` and the gates inside a circuit.
The `basis_gates` are what is supported by the backend. The circuit may have reset, and the backend may not support it (in fact this is what usually happens). So we want this pass to run on all *circuits* that have reset in them. 

The pass is correct. If you just remove this if statement, it should work. It will automatically not do anything for circuits that don't have reset (which is obviously correct, there is nothing to remove).

Reply:
Sure. I'm talking in the contest of the default pipeline. `RemoveResetInZeroState` is wasting cycles if there is no `Reset` in the circuit. If `RemoveResetInZeroState` runs after unrolling to the basis, then `Reset` needs to be in the basis for `RemoveResetInZeroState` to do something. This is only true in the particular situation of the this particular pipeline.

Reply:
No that's not true, just add this test and you'll see what I mean:
```
import math
from qiskit import QuantumCircuit, execute
circuit = QuantumCircuit(2, 2)

desired_vector = [1 / math.sqrt(2), 1 / math.sqrt(2), 0, 0]

circuit.initialize(desired_vector, range(0,2))
circuit.measure(range(0,2), range(0,2))
print(circuit)

from qiskit.test.mock import FakeMelbourne
from qiskit.compiler import transpile
device_backend = FakeMelbourne()
new_circuit = transpile(circuit, device_backend)
print(new_circuit)
```
Here the reset should be removed, but the way you've done it now leaves the reset in.
Reset is just ignored by the unroller if you look at the unroller code.

Reply:
you are right. I missed the introduction of `basic_insts`. I will remove the conditional. BTW, what this `TODO` means?
```python
            basic_insts = ['measure', 'reset', 'barrier', 'snapshot']
            if node.name in basic_insts:  # TODO: this is legacy behavior
                continue
```

Reply:
Done in 79fa7d71